examples/src/main/scala/zio/config/examples/CoproductExample.scala

```diff
@@ -15,7 +16,43 @@ object CoproductExample extends App {
   final case class Person(name: String, age: Option[Int])
   final case class Height(height: Long)
 
-  val danceConfig = description[Dance]
+  val personConfig =
+    (string("name") |@| int("age").optional)(Person.apply, Person.unapply)
+
+  val heightConfig =
```

Comment:
This is not the optimal way to do sum types.

First you should `orElseEither` `<+>` all of the terms together, and then xmap the whole thing. Then you have compile-time help if you add a new subtype—you'll be forced to deal with that. 

With this solution, if a new subtype is added, it will just not work at runtime, you won't get any runtime errors.

Reply:
@afsalthaj   gave me two possible solutions: see https://stackoverflow.com/questions/59670366/how-to-handle-an-adt-sealed-trait-with-zio-config/59690405#59690405

So you would suggest that Option 1 is the better one? Or do you have another one?